test/specs/enum_test.dart

```diff
@@ -73,4 +73,312 @@ void main() {
       }
     '''));
   });
+
+  test('should create an enum which mixes in and implements specs', () {
+    final myEnum = Enum((b) => b
+      ..name = 'MyEnum'
+      ..implements.addAll(const [
+        Reference('InterfaceA'),
+        Reference('InterfaceB'),
+      ])
+      ..mixins.addAll(const [
+        Reference('Mixin1'),
+        Reference('Mixin2'),
+      ])
+      ..values.addAll([
+        EnumValue((v) => v..name = 'a'),
+        EnumValue((v) => v..name = 'b'),
+        EnumValue((v) => v..name = 'c'),
+      ]));
+    expect(myEnum, equalsDart('''
+    enum MyEnum with Mixin1, Mixin2 implements InterfaceA, InterfaceB {
+      a,
+      b,
+      c
+    }
+    '''));
+  });
+
+  test('should create an enum which targets a named constructor', () {
+    final myEnum = Enum((b) => b
+      ..name = 'MyEnum'
+      ..constructors.addAll([
+        Constructor((c) => c..constant = true),
+        Constructor((c) => c
+          ..constant = true
+          ..name = 'named'),
+      ])
+      ..values.addAll([
+        EnumValue((v) => v..name = 'a'),
+        EnumValue((v) => v
+          ..name = 'b'
+          ..constructorName = 'named'),
+        EnumValue((v) => v..name = 'c'),
+      ]));
+    expect(myEnum, equalsDart('''
+    enum MyEnum {
+      a,
+      b.named(),
+      c;
+
+      const MyEnum();
+
+      const MyEnum.named();
+    }
+    '''));
+  });
+
+  test('should create an enum which targets a redirecting constructor', () {
+    final myEnum = Enum((b) => b
+      ..name = 'MyEnum'
+      ..constructors.addAll([
+        Constructor((c) => c..constant = true),
+        Constructor((c) => c
+          ..constant = true
+          ..name = 'redirect'
+          ..initializers.add(
+            refer('this').call([]).code,
+          )),
+      ])
+      ..values.addAll([
+        EnumValue((v) => v..name = 'a'),
+        EnumValue((v) => v
+          ..name = 'b'
+          ..constructorName = 'redirect'),
+        EnumValue((v) => v..name = 'c'),
+      ]));
+    expect(myEnum, equalsDart('''
+    enum MyEnum {
+      a,
+      b.redirect(),
+      c;
+
+      const MyEnum();
+
+      const MyEnum.redirect() : this();
+    }
+    '''));
+  });
+
+  test('should create an enum which targets a redirecting factory constructor',
+      () {
+    final myEnum = Enum((b) => b
+      ..name = 'MyEnum'
+      ..constructors.addAll([
+        Constructor((c) => c..constant = true),
+        Constructor((c) => c
+          ..constant = true
+          ..factory = true
```

Comment:
Oddly, const factory constructors are not allowed. See in [the spec](https://github.com/dart-lang/language/blob/master/accepted/future-releases/enhanced-enums/feature-specification.md):

> No-one is allowed to invoke a generative constructor and create an instance of the enum other than the enumerated enum values. That also means that a redirecting factory constructor cannot redirect to a generative constructor of an enum, and therefore no factory constructor of an enum declaration can be const, because a const factory constructor must redirect to a generative constructor.

I think we can keep the body of this constructor, unless you want to change it to something legal (could just always return `a` or throw or something...

Reply:
Interesting - good catch! @natebosch - what are your thoughts?

Reply:
I think it's probably fine to leave the test as is. I'm not overly worried about only demonstrating correct code in the tests, and there are plenty of ways we allow outputting invalid code.